java/arcs/core/allocator/Allocator.kt

```diff
@@ -77,11 +85,10 @@ class Allocator(val hostRegistry: HostRegistry) {
      * attaches generated keys.
      */
     private fun createStorageKeysIfNecessary(arcId: ArcId, idGenerator: Id.Generator, plan: Plan) =
-        plan.handleConnectionSpecs
-            .map { it -> it.handleSpec }
-            .filter { spec -> spec.storageKey == null }
-            .forEach { spec ->
-                spec.storageKey = createStorageKey(arcId, spec, idGenerator)
+        plan.particles
+            .map { it -> it.handles.values
```

Comment:
No need for `it ->`, that's implicit.

Also the `.filter` and `.forEach` should go outside the curly braces for the `.map`:

```
plan.particles
    .map { it.handles.values }
    .filter { spec -> spec.storageKey == null }
    .forEach { spec -> spec.storageKey = createStorageKey(arcId, idGenerator)}
```

(I would probably combine the filter and foreach into a single foreach with an if-statement, but doesn't really matter)